web-app/src/components/Legend/container.js

```diff
@@ -74,8 +74,15 @@ const LengendContainer = Wrapped =>
       };
     }
 
-    componentWillMount = () => {
-      this.createEmployeesList();
+    componentWillUpdate = (nextProps, nextState) => {
```

Comment:
Just a heads up react now declare `componentWillUpdate` as unsafe. From the top of my head I think they prefer to use `componentDidUpdate`.
https://reactjs.org/docs/react-component.html#unsafe_componentwillupdate